Snell/SnellCore/Router.swift

```diff
@@ -10,37 +10,63 @@ import Foundation
 
 public class Router {
   
-  public var routingTable:[NSRegularExpression:(request:Request) -> Response] = [:]
+  typealias Handler = (Request) -> () -> Response
+  
+  public enum HTTPMethod : String {
+    case Get     = "GET"
+    case Post    = "POST"
+    case Put     = "PUT"
+    case Delete  = "DELETE"
+    case Patch   = "PATCH"
+  }
+  
+  class Route {
+    var method:HTTPMethod
+    var pattern:NSRegularExpression
+    var handler:Handler
+    
+    init(method:HTTPMethod, pattern:NSRegularExpression, handler:Handler) {
+      self.method = method
+      self.pattern = pattern
+      self.handler = handler
+    }
+    
+    func matches(request:Request) -> Bool {
+      if (method.rawValue == request.method) {
+        let match = pattern.matchesInString(request.path as String, options: NSMatchingOptions(), range: NSMakeRange(0, request.path.characters.count))
+        return match.count > 0
+      }
+      return false
+    }
+  }
+  
+  var routingTable:[Route] = []
 
   public init() {
   }
-
-  public func route<T: Controller>(pattern:String, to action:(T) -> () -> Response) {
+  
+  public func route<T: Controller>(pattern:String, via:HTTPMethod = HTTPMethod.Get, to action:(T) -> () -> Response) {
     route(pattern, closure: { request in action(T(request: request))() })
   }
 
-  public func route(pattern:String, closure: (request:Request) -> Response) {
+  public func route(pattern:String, via:HTTPMethod = HTTPMethod.Get, closure: (request:Request) -> Response) {
     do {
       let regex = try NSRegularExpression(pattern: "^\(pattern)$", options: NSRegularExpressionOptions())
-      self.routingTable[regex] = closure
+      self.routingTable.append(Route(method: HTTPMethod.Get, pattern: regex, handler: { (r:Request) -> () -> Response in { closure(request: r) } }))
```

Comment:
The "via" method is not getting passed through here -- it's hardcoded as HTTPMethod.Get